lang/cs/Org.Apache.REEF.IO/FileSystem/AzureDataLake/SecretKeyAdlsCredentials.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+// 
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using System.Threading;
+using Microsoft.Rest;
+using Org.Apache.REEF.Tang.Annotations;
+using Microsoft.Rest.Azure.Authentication;
+using Org.Apache.REEF.IO.FileSystem.AzureDataLake.Parameters;
+
+namespace Org.Apache.REEF.IO.FileSystem.AzureDataLake
+{
+    internal sealed class SecretKeyAdlsCredentials : IAdlsCredentials
+    {
+        private readonly ServiceClientCredentials _credentials;
+
+        public ServiceClientCredentials Credentials
+        {
+            get { return _credentials; }
+        }
+
+        [Inject]
+        private SecretKeyAdlsCredentials([Parameter(typeof(Tenant))] string tenant,
+            [Parameter(typeof(TokenAudience))] string tokenAudience,
+            [Parameter(typeof(ClientId))] string clientId,
+            [Parameter(typeof(SecretKey))] string secretKey)
+        {
+            SynchronizationContext.SetSynchronizationContext(new SynchronizationContext());
```

Comment:
How the context is used?

Reply:
This is the recommended authentication code that is available on github which I have used
https://github.com/Azure-Samples/data-lake-analytics-dotnet-auth-options#getcreds_spi_secretkey